include/revng/Pipeline/Loader.h

```diff
@@ -211,10 +226,15 @@ class Loader {
 };
 } // namespace pipeline
 
-INTROSPECTION_NS(pipeline, ContainerDeclaration, Name, Type);
+INTROSPECTION_NS(pipeline, ContainerDeclaration, Name, Type, Role);
 template<>
-struct llvm::yaml::MappingTraits<pipeline::ContainerDeclaration>
-  : public TupleLikeMappingTraits<pipeline::ContainerDeclaration> {};
+struct llvm::yaml::MappingTraits<pipeline::ContainerDeclaration> {
+  static void mapping(IO &TheIO, pipeline::ContainerDeclaration &Info) {
+    TheIO.mapRequired("Name", Info.Name);
+    TheIO.mapOptional("Type", Info.Type);
+    TheIO.mapOptional("Role", Info.Role);
+  }
+};
```

Comment:
Why?

Reply:
before type was mandatory, now it is either type or role

Reply:
i am doing the current pattern in all other mapping in that file, should i switch over to the suggested pattern in all of them?

Reply:
Consider if it's worth it.
You could also consider switch to tuple trees.